test/es-module/test-import-map.mjs

```diff
@@ -0,0 +1,266 @@
+import tmpdir from '../common/tmpdir.js';
+import { describe, it } from 'node:test';
+import assert from 'node:assert';
+import path from 'node:path';
+import { execPath } from 'node:process';
+import { pathToFileURL } from 'node:url';
+import { writeFile } from 'node:fs/promises';
+import http from 'node:http';
+import import_map from 'internal/modules/esm/import_map';
+const { ImportMap } = import_map;
+import binding from 'internal/test/binding';
+const { primordials: { SafeMap, JSONStringify, ObjectEntries } } = binding;
+
+const importMapFixture = (...p) => fixtures.path('es-module-loaders', 'import-maps', ...p);
+const getImportMapPathURL = (name, filename = 'importmap.json') => {
+  return pathToFileURL(importMapFixture(name, filename));
+};
+const readImportMapFile = async (name, filename) => {
+  const url = getImportMapPathURL(name, filename);
+  return [url, (await import(url, { with: { type: 'json' } })).default];
+};
+const getImportMap = async (name, filename) => {
+  const [url, rawMap] = await readImportMapFile(name, filename);
+  return new ImportMap(rawMap, url);
+};
+const spawnPromisifiedWtihImportMap = async (name) => {
+  const entryPoint = importMapFixture(name, 'index.mjs');
+  const importMapPath = getImportMapPathURL(name).pathname;
+  const { code, signal, stdout, stderr } = await spawnPromisified(execPath, [
+    '--no-warnings',
+    // '--inspect', '--inspect-brk',
+    '--experimental-import-map', importMapPath,
+    entryPoint,
+  ], {
+    cwd: importMapFixture(name),
+  });
+  assert.strictEqual(code, 0, [stderr, stdout]);
+  assert.strictEqual(signal, null);
+  return stdout;
+}
+
+describe('Import Maps', { concurrency: true }, () => {
+  /**/
+  it('processImportMap - simple import map', async () => {
+    const importMap = await getImportMap('simple');
+    assert.deepStrictEqual(importMap.imports, new SafeMap(Object.entries({
+      foo: './foo/index.mjs'
+    })));
+    const expectedScopes = new SafeMap();
+    const fooScopeKey = new URL(importMap.baseURL, pathToFileURL('node_modules/foo'));
+    const fooScopeMap = new SafeMap(Object.entries({
+      bar: './baz.mjs'
+    }));
+    expectedScopes.set(fooScopeKey, fooScopeMap);
+    assert.deepStrictEqual(importMap.scopes, expectedScopes);
+  });
+
+  it('processImportMap - invalid import map', async () => {
+    await assert.rejects(
+      getImportMap('invalid'),
+      /^Error \[ERR_INVALID_IMPORT_MAP\]: Invalid import map: top level key "imports" is required and must be a plain object$/
+    );
+    await assert.rejects(
+      getImportMap('invalid', 'missing-scopes.json'),
+      /^Error \[ERR_INVALID_IMPORT_MAP\]: Invalid import map: top level key "scopes" is required and must be a plain object$/
+    );
+    await assert.rejects(
+      getImportMap('invalid', 'array-imports.json'),
+      /^Error \[ERR_INVALID_IMPORT_MAP\]: Invalid import map: top level key "imports" is required and must be a plain object$/
```

Comment:
We shouldn't check for error messages. We should only check for error codes.

Reply:
Good point, I can probably just drop the message part and only look for `ERR_INVALID_IMPORT_MAP`.